code/edge/src/test/java/com/adobe/marketing/mobile/EdgeStateTests.java

```diff
@@ -60,21 +63,192 @@ public void createSharedState(final Map<String, Object> state, final Event event
 	}
 
 	@Test
-	public void testBootupIfNeeded_LoadsPropertiesFromPersistence_AndCreatesSharedState_WithLocationHint() {
+	public void testBootupIfNeeded_LoadsPropertiesFromPersistence_AndCreatesSharedState_WithLocationHint()
+		throws InterruptedException {
+		CountDownLatch getSharedStateCalled = new CountDownLatch(1);
+		CountDownLatch createSharedStateCalled = new CountDownLatch(1);
+		mockSharedStateCallback =
+			new EdgeSharedStateCallback() {
+				@Override
+				public SharedStateResult getSharedState(final String stateOwner, final Event event) {
+					getSharedStateCalled.countDown();
+					if (!EdgeConstants.SharedState.HUB.equals(stateOwner)) {
+						return null;
+					}
+
+					return new SharedStateResult(SharedStateStatus.SET, new HashMap<>());
+				}
+
+				@Override
+				public void createSharedState(final Map<String, Object> state, final Event event) {
+					mockSharedState = state;
+					createSharedStateCalled.countDown();
+				}
+			};
+		state = new EdgeState(mockHitQueue, properties, mockSharedStateCallback);
+
 		long expectedExpiryTimestamp = setLocationHintInMockDataStore("or2", 100);
 		state.bootupIfNeeded();
+
 		assertEquals("or2", properties.getLocationHint());
+
+		assertTrue(getSharedStateCalled.await(200, TimeUnit.MILLISECONDS));
+		assertTrue(createSharedStateCalled.await(200, TimeUnit.MILLISECONDS));
 		assertNotNull(mockSharedState);
 		assertEquals("or2", mockSharedState.get(EdgeConstants.SharedState.Edge.LOCATION_HINT));
 	}
 
 	@Test
 	public void testBootupIfNeeded_LoadsPropertiesFromPersistence_AndCreatesSharedState_WithExpiredLocationHint()
 		throws InterruptedException {
+		CountDownLatch getSharedStateCalled = new CountDownLatch(1);
+		CountDownLatch createSharedStateCalled = new CountDownLatch(1);
+		mockSharedStateCallback =
+			new EdgeSharedStateCallback() {
+				@Override
+				public SharedStateResult getSharedState(final String stateOwner, final Event event) {
+					getSharedStateCalled.countDown();
+					if (!EdgeConstants.SharedState.HUB.equals(stateOwner)) {
+						return null;
+					}
+
+					return new SharedStateResult(SharedStateStatus.SET, new HashMap<>());
+				}
+
+				@Override
+				public void createSharedState(final Map<String, Object> state, final Event event) {
+					mockSharedState = state;
+					createSharedStateCalled.countDown();
+				}
+			};
+		state = new EdgeState(mockHitQueue, properties, mockSharedStateCallback);
 		long expectedExpiryTimestamp = setLocationHintInMockDataStore("or2", 1);
 		Thread.sleep(1100); // wait for hint to expire
 		state.bootupIfNeeded();
 		assertNull(properties.getLocationHint());
+
+		assertTrue(getSharedStateCalled.await(200, TimeUnit.MILLISECONDS));
```

Comment:
What are your thoughts on using mockito to mock the shared state callback? I think that way you will not need to wait for the 200ms to and also make tests simpler. 

```
@Mock
private EdgeSharedStateCallback mockSharedStateCallback;
...

when(mockSharedStateCallback.getSharedState(EdgeConstants.SharedState.HUB, any()))
			.thenReturn(new SharedStateResult(SharedStateStatus.SET, Collections.EMPTY_MAP));		

state = new EdgeState(mockHitQueue, properties, mockSharedStateCallback)
state.bootupIfNeeded();

ArgumentCaptor<..> stateCaptor = ArgumentCaptor.forClass(...);
verify(mockSharedStateCallback).createSharedState(stateCaptor.capture(), ..);
assertTrue(stateCaptor.getValue().isEmpty());
```

Reply:
Seems doable, in fact I am using a similar approach when mocking the extensionApi for shared state usages in the EdgeExtensionTests in PR #10. Since I stacked some of the changes on top of this, I can take this change in a separate PR if that's ok.